src/org/rascalmpl/library/Prelude.java

```diff
@@ -3339,50 +3368,67 @@ private void copy(Reader from, Writer to) throws IOException {
 		}
 	}
 
-
-	private String toBase64(InputStream src, int estimatedSize) throws IOException {
+	private String toBase64(InputStream src, int estimatedSize, boolean includePadding) throws IOException {
 	  ByteArrayOutputStream result = new ByteArrayOutputStream(estimatedSize);
-	  OutputStream encoder = Base64.getEncoder().wrap(result);
-	  copy(src, encoder);
-	  encoder.close();
+	  Encoder encoder = Base64.getEncoder();
+	  if (!includePadding) {
+	  	encoder = encoder.withoutPadding();
+	  }
+	  OutputStream dest = encoder.wrap(result);
+	  copy(src, dest);
+	  dest.close();
 	  return result.toString(StandardCharsets.ISO_8859_1.name());
 	}
 
-	public IString toBase64(IString in) {
+	public IString toBase64(IString in, IString charsetName, IBool includePadding) {
 	  try {
-	      InputStream bytes = new ByteBufferBackedInputStream(StandardCharsets.UTF_8.encode(in.getValue()));
-	      return values.string(toBase64(bytes, in.length() * 2));
+		Charset charset = Charset.forName(charsetName.getValue());
+	    InputStream bytes = new ByteBufferBackedInputStream(charset.encode(in.getValue()));
+	    return values.string(toBase64(bytes, in.length() * 2, includePadding.getValue()));
 	  } catch (IOException e) {
 	      throw RuntimeExceptionFactory.io(values.string(e.getMessage()));
 	  }
 	}
 
-	public IString toBase64(ISourceLocation file) {
-	    try (InputStream in = REGISTRY.getInputStream(file)) {
-	        return values.string(toBase64(in, 1024));
-	    }
-	    catch (IOException e) {
-	        throw RuntimeExceptionFactory.io(values.string(e.getMessage()));
-	    }
+	public IString toBase64(ISourceLocation file, IBool includePadding) {
+		return readBase64(file, includePadding);
 	}
-	
-	
 
 	private void fromBase64(String src, OutputStream target) throws IOException {
 	  InputStream bytes = new ByteBufferBackedInputStream(StandardCharsets.ISO_8859_1.encode(src));
 	  copy(Base64.getDecoder().wrap(bytes), target);
 	}
 
-	public IString fromBase64(IString in) {
+	public IString fromBase64(IString in, IString charset) {
 	    try {
 	        ByteArrayOutputStream result = new ByteArrayOutputStream(in.length());
 	        fromBase64(in.getValue(), result);
-	        return values.string(result.toString(StandardCharsets.UTF_8.name()));
+	        return values.string(result.toString(charset.getValue()));
 	    } catch (IOException e) {
 	        throw RuntimeExceptionFactory.io(values.string(e.getMessage()));
 	    }
 	}
 
+	public IString toBase32(IString in, IString charset, IBool includePadding) {
+		Base32 encoder = new Base32();
+		ByteBuffer buffer = Charset.forName(charset.getValue()).encode(in.getValue());
+		byte[] data = new byte[buffer.limit()];
```

Comment:
can we not use `String.getBytes(Charset)` directly?

Reply:
Fixed.